scvi/model/_destvi.py

```diff
@@ -0,0 +1,282 @@
+    >>> gamma = spatial_model.get_gamma(st_adata)
+    """
+
+    def __init__(
+        self,
+        st_adata: AnnData,
+        cell_type_mapping: np.ndarray,
+        decoder_state_dict: OrderedDict,
+        px_decoder_state_dict: OrderedDict,
+        px_r: np.ndarray,
+        n_hidden: int,
+        n_latent: int,
+        n_layers: int,
+        **module_kwargs,
+    ):
+        st_adata.obs["_indices"] = np.arange(st_adata.n_obs)
+        register_tensor_from_anndata(st_adata, "ind_x", "obs", "_indices")
+        super(DestVI, self).__init__(st_adata)
+        self.module = MRDeconv(
+            n_spots=st_adata.n_obs,
+            n_labels=cell_type_mapping.shape[0],
+            decoder_state_dict=decoder_state_dict,
+            px_decoder_state_dict=px_decoder_state_dict,
+            px_r=px_r,
+            n_genes=st_adata.n_vars,
+            n_latent=n_latent,
+            n_layers=n_layers,
+            n_hidden=n_hidden,
+            **module_kwargs,
+        )
+        self.cell_type_mapping = cell_type_mapping
+        self._model_summary_string = "DestVI Model"
+        self.init_params_ = self._get_init_params(locals())
+
+    @classmethod
+    def from_rna_model(
+        cls,
+        st_adata: AnnData,
+        sc_model: CondSCVI,
+        **model_kwargs,
+    ):
+        """
+        Alternate constructor for exploiting a pre-trained model on RNA-seq data.
+
+        Parameters
+        ----------
+        st_adata
+            registed anndata object
+        sc_model
+            trained CondSCVI model
+        use_gpu
+            Use the GPU or not.
+        **model_kwargs
+            Keyword args for :class:`~scvi.model.DestVI`
+        """
+        decoder_state_dict = sc_model.module.decoder.state_dict()
+        px_decoder_state_dict = sc_model.module.px_decoder.state_dict()
+        px_r = sc_model.module.px_r.detach().cpu().numpy()
+
+        return cls(
+            st_adata,
+            sc_model.scvi_setup_dict_["categorical_mappings"]["_scvi_labels"][
+                "mapping"
+            ],
+            decoder_state_dict,
+            px_decoder_state_dict,
+            px_r,
+            sc_model.module.n_hidden,
+            sc_model.module.n_latent,
+            sc_model.module.n_layers,
+            **model_kwargs,
+        )
+
+    def get_proportions(
+        self, adata: AnnData = None, keep_noise: bool = False
+    ) -> np.ndarray:
+        """
+        Returns the estimated cell type proportion for the spatial data.
+
+        Shape is n_cells x n_labels OR n_cells x (n_labels + 1) if keep_noise.
+
+        Parameters
+        ----------
+        adata
+            the anndata to loop through (used only if amortization)
+        keep_noise
+            whether to account for the noise term as a standalone cell type in the proportion estimate.
+        """
+        if self.is_trained_ is False:
+            raise RuntimeError("Please train the model first.")
```

Comment:
we need to change this in the codebase... all cases of this should be user warning that model isn't trained.